nginx.config/default.conf

```diff
@@ -0,0 +1,10 @@
+server {
+    listen 80;
+
+    location / {
+        proxy_buffering off;
+        root /usr/share/nginx/html;
+        try_files $uri $uri/ $uri/index.html /index.html =404;
```

Comment:
@indraraj how should 404s be handled?

Reply:
Will remove it since you don't have 404 html file in the `dist` dir.

Reply:
its a  single page application we don't need a separate 404 html, 404 scenarios are already handled in UI via routes

Reply:
@yahyaAlsaidi I know there is the original `50x.html` file. But I am not sure about 404. I need to check.

Reply:
No, I delete it this part when merging the two commits together, you don't have to check it there is not 404 file there. Sometimes some bundler include that and some not so you don't have it, so there is no reason to add it.

Reply:
Ah yes, I double-checked. There's indeed no 404.html file and the UI frontend is supposed to handle 404s.

One thing that was coming to my mind and I am interested to hear your opinion!
What do you think about using a different directory inside the container to host the UI?
Like `/srv/` or a new directory like `dbz-ui` instead of  `/usr/share/nginx/html`? And then it makes really sense to create a new config file in `conf.d/` dir for that endpoint with your config? That way we done interfere with the default config but we need to try if this will be working. wdyt?

Reply:
Well, simply including your custom configuration in the default file might not resolve the issue due to potential conflicts. Some opt for a different approach by organizing configurations into sites-available and sites-enabled directories. Configurations are placed in sites-available and then symlinked in sites-enabled, which can be included in nginx.conf. However, this method can add complexity, especially if accidental symlink or backup files are created. You can find more discussion on this topic [here](https://serverfault.com/questions/527630/difference-in-sites-available-vs-sites-enabled-vs-conf-d-directories-nginx). While separation of concerns is important, sometimes simplicity is preferable. Also [here](https://antonlytvynov.medium.com/nginx-sites-available-sites-enabled-3bd025bc4d25) kind of example to do something similar to that.

Reply:
You should read the second answer in the first link (Stackoverflow). I think that one is very accurate `Best Practice Is conf.d.` and `You should be using /etc/nginx/conf.d, as that's a standard convention, and should work anywhere.` It think this reply is really on point. In a dockerized environment it does not make sense to use `sites-available`/`sites-enable`. I think this pattern only makes sense when you have one big server hosting multiple different websites/customers etc.

So that leaves the question if we should overwrite the upstream `conf.d/default.conf` or add an additional `conf.d/debezium-ui.conf`? I lean towards the second options if `default.conf` and `debezium-ui.conf` dont't interfere and can work together without harm. Maybe we only need to add the `try_files` line inside the correct structure in `debezium-ui.conf`?

Reply:
You could give it a try.